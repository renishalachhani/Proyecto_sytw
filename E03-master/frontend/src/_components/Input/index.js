import React from 'react'

export default class Input extends React.Component {
  render() {
    return (
      <div>
        <input
          data-testid="input"
          name={this.props.name}
          ref={this.props.ref}
          type={this.props.type}
          value={this.props.value}
          onChange={this.props.onChange}
        />
      </div>
    )
  }
}
