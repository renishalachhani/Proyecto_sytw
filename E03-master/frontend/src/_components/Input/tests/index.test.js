import React from 'react'
import jsdom from 'jsdom'
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Input from '../index'

configure({ adapter: new Adapter() })

const { JSDOM } = jsdom
const dom = new JSDOM('<!doctype html><html><body></body></html>')
const { window } = dom

global.window = window
global.document = window.document

const name = 'Prueba'
const type = 'otraPrueba'
const value = 'valorPrueba'
const change = jest.fn()
const renderComponent = (props = {}) =>
  render(
    <Input name={name} type={type} value={value} onChange={change} {...props} />
  )

describe('<Input />', () => {
  afterEach(cleanup)

  it('Render', () => {
    const div = document.createElement(div)
    ReactDOM.render(<Input />, div)
  })

  it('Should have a name attribute', () => {
    const { container } = renderComponent()
    const element = container.querySelector('input')
    expect(element.hasAttribute('name')).toBe(true)
  })

  it('Should have a type attribute', () => {
    const { container } = renderComponent()
    const element = container.querySelector('input')
    expect(element.hasAttribute('type')).toBe(true)
  })

  it('Should have a value attribute', () => {
    const { container } = renderComponent()
    const element = container.querySelector('input')
    expect(element.hasAttribute('value')).toBe(true)
  })

  it('Matches snapshots', () => {
    let app
    const tree = renderer
      .create(
        <Input
          name={name}
          ref={(inst) => {
            app = inst
          }}
          type={type}
          value={value}
          onChange={change}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
