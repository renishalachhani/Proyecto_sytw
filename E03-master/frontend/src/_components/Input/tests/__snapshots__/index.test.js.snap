// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Matches snapshots 1`] = `
<div>
  <input
    data-testid="input"
    name="Prueba"
    onChange={[MockFunction]}
    type="otraPrueba"
    value="valorPrueba"
  />
</div>
`;
